middleware/http/opa/middleware.go

```diff
@@ -43,6 +46,26 @@ var (
 	errOpaInvalidResultType = errors.New("got an invalid type back from repo policy. Only a boolean or map is valid")
 )
 
+func (s *Status) UnmarshalJSON(b []byte) error {
+	if len(b) == 0 {
+		return nil
+	}
+	if b[0] == '"' {
```

Comment:
If you want only unmarshal the `DefaultStatus` as a custom type, I would suggest using the typesafe handling of go instead of attempting to infer the type by walking the array (for example, this code doesn't account for a byte array with leading `"` but not ending `"`).

```go
func (s *Status) UnmarshalJSON(b []byte) error {
	var v interface{}
	if err := json.Unmarshal(b, &v); err != nil {
		return err
	}
	switch t := v.(type) {
	case string:
		if i, err := strconv.Atoi(t); err != nil {
			return err
		} else {
			*s = Status(i)
		}
	case int:
		*s = Status(t)
	default:
		return fmt.Errorf("invalid defaultStatus value: %v", t)
	}
	return nil
}
```